app/src/main/java/com/github/geohunt/app/ui/screens/teamprogress/TeamProgressScreenContent.kt

```diff
@@ -0,0 +1,130 @@
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.model.Location
+import com.github.geohunt.app.model.User
+import com.github.geohunt.app.ui.components.bounties.BountyChallengeCard
+import com.github.geohunt.app.ui.components.teamprogress.TeamProgressMembersCarousel
+import com.github.geohunt.app.ui.components.teamprogress.TeamProgressTopAppBar
+import com.github.geohunt.app.ui.utils.pagination.FinitePagedList
+import kotlinx.coroutines.flow.StateFlow
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+fun TeamProgressScreenContent(
+    onBack: () -> Unit,
+    onLeaderboard: () -> Unit,
+    onChat: () -> Unit,
+    onClaim: (Challenge) -> Unit,
+
+    teamName: String,
+    teamMembers: FinitePagedList<User>,
+    claimState: FinitePagedList<Boolean>,
+    newMessages: StateFlow<Int>,
+    locationState: StateFlow<Location?>,
+    challenges: List<Challenge>?
+) {
+    val scrollBehavior = TopAppBarDefaults.exitUntilCollapsedScrollBehavior(rememberTopAppBarState())
+    val currentLocation = locationState.collectAsStateWithLifecycle()
+
+    Scaffold(
+        modifier = Modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
+        topBar = { TeamProgressTopAppBar(
+            // Try to limit team name to 20 characters, otherwise it is truncated in the app bar,
+            // which is generally bad UX. The current work-around is to have ellipsis (...)
+            // when the team name is too long.
+            teamName = teamName,
+            onBack = onBack,
+            onLeaderboard = onLeaderboard,
+            onChat = onChat,
+            newMessagesState = newMessages,
+            scrollBehavior = scrollBehavior
+        ) }
+    ) { padding ->
+        LazyColumn(
+            Modifier
+                .padding(padding)
+                .fillMaxWidth()) {
+            item {
+                Column(Modifier.padding(horizontal = 16.dp)) {
+                    Text(
+                        "${teamMembers.size()} members",
+                        style = MaterialTheme.typography.titleLarge
+                    )
+
+                    TeamProgressMembersCarousel(teamMembers = teamMembers)
+                }
+            }
+
+            stickyHeader {
+                Surface {
+                    Row(
+                        modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        Text(
+                            "Challenges",
+                            style = MaterialTheme.typography.titleLarge
+                        )
+
+                        Spacer(Modifier.weight(1.0f))
+
+                        /*Text(
```

Comment:
Comment that should not be there?

Reply:
This was mainly to remind that there should be some indicator of the progression here, but did not have the time to do any because I'm overworked 👍